core/src/test/scala/unit/kafka/utils/JsonTest.scala

```diff
@@ -62,5 +65,26 @@ class JsonTest {
     assertEquals("{\"a\":1,\"b\":2}", Json.encode(Map("a" -> 1, "b" -> 2)))
     assertEquals("{\"a\":[1,2],\"c\":[3,4]}", Json.encode(Map("a" -> Seq(1,2), "c" -> Seq(3,4))))
   }
-  
+
+  @Test
+  def testParseTo() = {
+    val foo = "baz"
+    val bar = 1
+
+    val result = Json.parseTo(s"""{"foo": "$foo", "bar": $bar}""", classOf[TestObject])
+
+    assertTrue(result.isRight)
+    assertEquals(TestObject(foo, bar), result.right.get)
```

Comment:
Maybe we can also have a test where we can roundtrip (serialize and then deserialize)? It would be nice to have a method in `Json` to serialize with `Jackson`. Maybe we can even actually replace the existing one.